test/small_granularity.c

```diff
@@ -0,0 +1,126 @@
+#include <small/small.h>
+#include <small/quota.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include "unit.h"
+
+enum {
+	GRANULARITY_MIN = 4,
+	GRANULARITY_MAX = 1024,
+	ALLOCATION_COUNT = 10,
+};
+
+struct slab_arena arena;
+struct slab_cache cache;
+struct small_alloc alloc;
+struct quota quota;
+
+static unsigned *ptrs[FACTOR_POOL_MAX];
+
+static inline void
+free_checked(struct mempool *pool, unsigned *ptr)
+{
+	/*
+	 * Checking previous saved data.
+	 */
+	fail_unless(ptr[0] < FACTOR_POOL_MAX &&
+		    ptr[pool->objsize/sizeof(unsigned)-1] == ptr[0]);
+	int pos = ptr[0];
+	fail_unless(ptrs[pos] == ptr);
+	ptrs[pos][0] = ptrs[pos][pool->objsize/sizeof(unsigned)-1] = INT_MAX;
+	mempool_free(pool, ptrs[pos]);
+	ptrs[pos] = NULL;
+}
+
+static inline void *
+alloc_checked(struct mempool *pool, unsigned pos)
+{
+	fail_unless(ptrs[pos] == NULL);
+	ptrs[pos] = mempool_alloc(pool);
+	fail_unless(ptrs[pos] != NULL);
+	/*
+	 * Saving some data for test purposes
+	 */
+	ptrs[pos][0] = pos;
+	ptrs[pos][pool->objsize/sizeof(unsigned)-1] = pos;
+	return ptrs[pos];
+}
+
+static void
+small_granularity_test(void)
+{
+	header();
+	const float alloc_factor = 1.3;
+	for(unsigned int granularity = GRANULARITY_MIN;
+	    granularity <= GRANULARITY_MAX;
+	    granularity <<= 1) {
+	    	float actual_alloc_factor;
+		/*
+		 * Creates small_alloc with different granularity,
+		 * which must be power of two.
+		 */
+		small_alloc_create(&alloc, &cache, granularity,
+			   granularity, alloc_factor, &actual_alloc_factor);
+		/*
+		 * Checks aligment of all mempools in small alloc.
```

Comment:
Nit: maybe extract this code and code related to `Checks allocation of all mempools in small_alloc` into two separate functions? Because they're two independent tests.